### PR TITLE
New logic to provide database with a key stored in a local file

### PR DESCRIPTION
**Pull request by Aurora12**
See `supplier.go:setupConnection()`.

The new setting in the server configuration is `SqlSettings.Security` which takes one of the three values: `NONE` (or empty/absent), `LOCAL` (means local file) or `VAULT`.

When the setting says `LOCAL` or `VAULT` the respective files required for them to work MUST be available on the local file system. For `LOCAL` that is a text file with a key stored at `config.go:DATABASE_SECURITY_LOCAL_FILE`, for `VAULT` that is a text file with a k8s access token stored at `vault.go:VAULT_K8S_SERVICE_ACCOUNT_TOKEN_FILE`.

The `IVault` no longer has anything to do with sending the key to the db. Its logic became more abstract and allows for getting any values from the Vault (we'll need more later).

### CODE REVIEW

File: store/sqlstore/vault.go [hunk 1]
```diff
@@ -102,29 +99,17 @@ type VaultKVSecret struct {
...
-	}
-	file, err := os.Open(name)
-	if err != nil {
-		mlog.Warn("Cannot open k8s token file", mlog.String("k8s token file", name))
-		return "", errors.Wrap(err, "Cannot open token file")
-	}
-	defer file.Close()
-	token, err := ioutil.ReadAll(file)
+func (*Vault) Getk8sServiceAccountToken(name string) (*string, error) {
+	token, err := utils.ReadTextFile(name)
```
Comment (ygworldr) [inline store/sqlstore/vault.go:108]: Wish I knew/remembered about that function!!!

File: utils/utils.go [hunk 2]
```diff
@@ -107,3 +112,25 @@ func GetHostnameFromSiteURL(siteURL string) string {
 
 	return u.Hostname()
 }
+
+// ReadTextFile reads a whole file content as a string and wraps basic errors.
+func ReadTextFile(name string) (*string, error) {
```
Comment (ygworldr) [inline utils/utils.go:117]: Ah, that's why I did not know about it! :grin:

**Review by ygworldr — APPROVED** [review]
LGTM

File: store/sqlstore/vault.go (hunk 1 above)
Comment (Aurora12) [reply store/sqlstore/vault.go:108]: I've created it by copying your code from vault.go 😁
It doesn't belong there as there is nothing vault-specific in reading a text file.

**Pull request merged**